tests/wallclock/tst_wallclock.cpp

```diff
@@ -82,20 +82,13 @@ void tst_WallClock::update()
     timezoneAbbreviationSpy.clear();
     timezoneOffsetSpy.clear();
 
-    QDateTime currentTime = QDateTime::currentDateTime();
-    int year = obj->property("year").toInt();
-    int month = obj->property("month").toInt();
-    int day = obj->property("day").toInt();
-    int hour = obj->property("hour").toInt();
-    int minute = obj->property("minute").toInt();
-    int second = obj->property("second").toInt();
-    QCOMPARE(year, currentTime.date().year());
-    QCOMPARE(month, currentTime.date().month()-1);
-    QCOMPARE(day, currentTime.date().day());
-    QCOMPARE(hour, currentTime.time().hour());
-    QCOMPARE(minute, currentTime.time().minute());
-    QCOMPARE(second, currentTime.time().second());
-    int lastSecond = second;
+    int currentSecs = QDateTime::currentDateTime().toMSecsSinceEpoch() / 1000;
+
+    QDateTime time = obj->property("time").toDateTime();
+    QVERIFY(time.isValid());
+    int timeSecs = time.toMSecsSinceEpoch() / 1000;
+    // giving some leeway
+    QVERIFY(currentSecs == timeSecs || currentSecs == timeSecs + 1);
```

Comment:
I see no reason why this would not work, but ...

There is toSecsSinceEpoch() method that could be used instead of dividing millsecs by 1000.

Or, depending on is jitter expected only in one or both directions, one could perhaps use some form of QVERIFY(abs(deltaInMs) <= 1000) type thing.

Reply:
["This function was introduced in Qt 5.8."](https://doc.qt.io/qt-5/qdatetime.html#toSecsSinceEpoch) :thinking:

But I like the idea of using abs() there.

Reply:
One slight problem with abs(), even if small, is that it wouldn't guarantee the ordering of the timestamps.

Reply:
Since this is approved, I'll just merge :)